src/porepy/grids/grid_bucket.py

```diff
@@ -728,6 +728,85 @@ def eliminate_node(self, node):
 
 #------------------------------------------------------------------------------#
 
+    def loop_nodes(self, fct):
```

Comment:
1) Can't this be achieved by values = [fct(g, d) for g, d in gb]
2) If you still want to keep the function, the name should reflect that you evaluate a function on all nodes